### PR TITLE
[lit-next] Pre-release READMEs

### LINKED ISSUE #1262: [lit-next] Update READMEs
We need new READMEs for the monorepo, lit-html and lit-element.

lit-html and lit-element READMEs will be on npm and our only documentation for now. They should describe the major changes, the state and intended audience of the preview release, and point to change logs and the 2.0/3.0 Ideas issues.

### PR DESCRIPTION
**Pull request by kevinpschaaf**
Fixes #1262

### CODE REVIEW

File: packages/lit-html/README.md [hunk 1]
```diff
@@ -1,14 +1,133 @@
...
+| | 1.x API | 2.0 API |
+|-|-----|-----|
+| Code idiom for directive | function that takes directive arguments, and returns function that takes `part` and returns value | class with `update` & `render` methods which accept directive arguments |
+| Where to do declarative rendering | pass value to `part.setValue()` | return value from `render()` method |
+| Where to do imperative DOM/part manipulation | directive function | `update()` method |
+| Where state is stored between renders | `WeakMap` keyed on `part` | class instance fields |
+| How part validation is done | `instanceof` check on `part` in every render | `part.type` check in constructor
+
+### Example directive migration
+Below is an example of a fictional directive, and how to migrate it to the new
```
Comment (sorvell) [inline packages/lit-html/README.md:66]:
```suggestion
Below is an example of a functional directive, and how to migrate it to the new
```

File: packages/lit-html/README.md [hunk 2]
```diff
@@ -1,14 +1,133 @@
...
+  // Update state
+  if (previous === undefined) {
+    value = initialValue;
+  } else {
+    value++;
+  }
+  // Store state
+  part.setValue(value);
+  // Update part with new rendering
+  previousState.set(part, html`<p>${value}</p>`);
```
Comment (sorvell) [inline packages/lit-html/README.md:93]: Shouldn't this be:
```
// Store state
previousState.set(part, value);
// Update part with new rendering
part.setValue(html`<p>${value}</p>`);
```

File: packages/lit-html/README.md [hunk 3]
```diff
@@ -1,14 +1,133 @@
...
+  * `until`
+
+## 🚨 Migrating directives
+
+While the API for _using_ directives should be 100% backward-compatible with
+1.x, there is a breaking change to how custom directives are _authored_. The API
+change improves ergonomics around making stateful directives while providing a
+clear pattern for SSR-compatible directives: only `render` will be called on the
+server, while `update` will not be.
+
```
Comment (sorvell) [inline packages/lit-html/README.md:52]: Add: `WARNING: The directive API is experimental and subject to change.`

**Review by sorvell — CHANGES_REQUESTED** [review]

File: packages/lit-html/README.md (hunk 1 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md:66]: That was actually intended to be _fiction_al... (i.e. aka it's made up, not a useful thing). Is that a weird way to say it?

File: packages/lit-html/README.md (hunk 2 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md:93]: Whoops! Thank you.

File: packages/lit-html/README.md (hunk 3 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md:55]: Done

File: packages/lit-html/README.md (hunk 2 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md]: Updated

File: packages/lit-html/README.md [hunk 4]
```diff
@@ -1,14 +1,140 @@
...
+**⚠️ WARNING: The directive and part API changes are experimental and subject to
```
Comment (graynorton) [inline packages/lit-html/README.md:55]: I feel like what we're aiming for here is less "experimental" and more "work in progress", right? As it reads right now, it almost sounds like we might revert to the previous API if the "experiment" doesn't work out.

File: packages/lit-html/README.md [hunk 5]
```diff
@@ -1,14 +1,140 @@
...
+  // Store state
+  previousState.set(part, value);
+  // Update part with new rendering
+  part.setValue(html`<p>${value}</p>`);
+});
+```
+
+2.0 Directive API:
+```js
+import {directive, Directive, NODE_PART} from 'lit-html';
```
Comment (graynorton) [inline packages/lit-html/README.md:104]: You still need to import `html` here, right?

File: README.md [hunk 6]
```diff
@@ -1,3 +1,42 @@
 # lit-html 2.0 monorepo
 
-A collection of packages related to lit-html 2.0 work
+A collection of packages related to lit-html 2.0 work.
```
Comment (justinfagnani) [inline README.md:3]:
```suggestion
A collection of packages related to lit-html 2.0 and LitElement 3.0 work.
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,42 @@
...
+following links for details on the changes being made:
+* [Ideas for lit-html 2.0](https://github.com/Polymer/lit-html/issues/1182)
+* [Ideas for LitElement 3.0](https://github.com/Polymer/lit-element/issues/1077)
+* [lit-next issues/PRs in github](https://github.com/Polymer/lit-html/issues?q=is%3Aissue+label%3Alit-next+)
+
+## Packages
+
+* [`lit-html`](./packages/lit-html)
+* [`lit-element`](./packages/lit-element)
+* [`benchmarks`](./packages/benchmarks)
```
Comment (justinfagnani) [inline README.md:15]: Add `./packages/tests`?

File: packages/lit-html/README.md (hunk 1 above)
Comment (justinfagnani) [reply packages/lit-html/README.md:71]: "fictional functional"? :)

Comment (justinfagnani) [reply packages/lit-html/README.md:71]: More seriously, "example of a lit-html 1.x directive"

**Review by justinfagnani — APPROVED** [review]
LGTM

File: packages/lit-html/README.md (hunk 5 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md:104]: Yes, thanks!

File: packages/lit-html/README.md (hunk 4 above)
Comment (kevinpschaaf) [reply packages/lit-html/README.md]: Sure, done!

File: README.md (hunk 6 above)
Comment (kevinpschaaf) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (kevinpschaaf) [reply README.md:16]: Done

**Review by sorvell — APPROVED** [review]

**Pull request merged**